### PR TITLE
Clean up the Makefile in src/runtime/hexagon

### PR DESCRIPTION
**Pull request by steven-johnson**
Add documentation; have "make all" do the right thing by building everything (assume env vars are correct); redo binary output so that all build products are in a single hierarchy rather than spread across various places.

### CODE REVIEW

File: src/runtime/hexagon_remote/bin/v60/halide_hexagon_remote.so.map [hunk 1]
```diff
@@ -0,0 +1,126 @@
+
+Linker Script and memory map
```
Comment (dsharletg) [inline src/runtime/hexagon_remote/bin/v60/halide_hexagon_remote.so.map]: I don't think this file needs to be added to the repo, was it intentional to add it?

Comment (steven-johnson) [inline src/runtime/hexagon_remote/bin/v60/halide_hexagon_remote.so.map]: Removed

**Comment by dsharletg** [comment]
Thanks for the cleanup and docs especially. I'm going to test this on device real quick, since the IDL compiler generated something different.

**Comment by dsharletg** [comment]
Works on device too, merging this now.

@pranavb-ca @ronlieb @dpalermo FYI, the build for the hexagon remote has moved around a bit.

**Pull request merged**